src/router/routes/deploy.js

```diff
@@ -0,0 +1,132 @@
+//
+// SecureImage
+//
+// Copyright © 2018 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Created by Jason Leach on 2018-01-10.
+//
+
+/* eslint-env es6 */
+
+'use strict';
+
+import * as minio from 'minio';
+import url from 'url';
+import util from 'util';
+import request from 'request-promise-native';
+import { Router } from 'express';
+import config from '../../config';
+import { logger } from '../../libs/logger';
+import {
+  asyncMiddleware,
+  errorWithCode,
+} from '../../libs/utils';
+import {
+  createBucketIfRequired,
+  bucketExists,
+  isExpired,
+} from '../../libs/bucket';
+import DataManager from '../../libs/db';
+
+const router = new Router();
+const dm = new DataManager();
+const {
+  db,
+  Job,
+} = dm;
+const bucket = config.get('minio:bucket');
+const client = new minio.Client({
+  endPoint: config.get('minio:endPoint'),
+  port: config.get('minio:port'),
+  secure: config.get('minio:secure'),
+  accessKey: config.get('minio:accessKey'),
+  secretKey: config.get('minio:secretKey'),
+  region: config.get('minio:region'),
+});
+
+try {
```

Comment:
You can remove this check. I'll move it to get done in `index.js` so its only done one time.